Sources/NIOHPACK/IndexedHeaderTable.swift

```diff
@@ -0,0 +1,150 @@
+    /// - Returns: A tuple containing the index of any located header, and a boolean indicating
+    ///            whether the item at that index also contains a matching value. Returns `nil`
+    ///            if no match could be found.
+    public func firstHeaderMatch(for name: String, value: String?) -> (index: Int, matchesValue: Bool)? {
+        return self.firstHeaderMatch(for: name.utf8, value: value?.utf8)
+    }
+    
+    func firstHeaderMatch<C : Collection>(for name: C, value: C?) -> (index: Int, matchesValue: Bool)? where C.Element == UInt8 {
+        var firstHeaderIndex: Int? = nil
+        for index in self.staticTable.indices(matching: name) {
+            // we've found a name, at least
+            if firstHeaderIndex == nil {
+                firstHeaderIndex = index
+            }
+            
+            if let value = value, self.staticTable.view(of: self.staticTable[index].value).matches(value) {
+                return (index, true)
+            }
+        }
+        
+        // no complete match: search the dynamic table now
+        if let result = self.dynamicTable.findExistingHeader(named: name, value: value) {
+            if let staticIndex = firstHeaderIndex, result.containsValue == false {
+                // Dynamic table can't match the value, and we have a name match in the static
+                // table. In this case, we prefer the static table.
+                return (staticIndex, false)
+            } else {
+                // Either no match in the static table, or the dynamic table has a header with
+                // a matching value. Return that, but update the index appropriately.
+                return (result.index + StaticHeaderTable.count, result.containsValue)
+            }
+        } else if let staticIndex = firstHeaderIndex {
+            // nothing in the dynamic table, but the static table had a name match
+            return (staticIndex, false)
+        } else {
+            // no match anywhere, you'll have to encode the whole thing
+            return nil
+        }
+    }
+    
+    /// Appends a header to the table.
+    ///
+    /// This call may result in an empty table, as per RFC 7541 § 4.4:
+    /// > "It is not an error to attempt to add an entry that is larger than the maximum size;
+    /// > an attempt to add an entry larger than the maximum size causes the table to be
+    /// > emptied of all existing entries and results in an empty table."
+    ///
+    /// - Parameters:
+    ///   - name: The name of the header to insert.
+    ///   - value: The value of the header to insert.
+    /// - Returns: `true` if the header was added to the table, `false` if not.
+    public mutating func add(headerNamed name: String, value: String) throws {
+        try self.dynamicTable.addHeader(named: name.utf8, value: value.utf8)
+    }
+    
+    public mutating func add<C : ContiguousCollection>(headerNameBytes nameBytes: C, valueBytes: C) throws where C.Element == UInt8 {
```

Comment:
This is a bit over specified, as noted before. Also: remove the whitespace before the colon.